File: upload_to_openai.js
const fs = require('fs');
const fetch = require('node-fetch');
const FormData = require('form-data');

const uploadFiles = async () => {
  const openaiApiKey = process.env.OPENAI_API_KEY;
  const buildDir = 'build';

  fs.readdir(buildDir, (err, files) => {
    if (err) {
      console.error('Error reading the build directory:', err);
      return;
    }

    files.forEach(file => {
      const filePath = `${buildDir}/${file}`;
      if (fs.statSync(filePath).isFile()) {
        const formData = new FormData();
        formData.append('file', fs.createReadStream(filePath));
        formData.append('purpose', 'code_review');

        fetch('https://api.openai.com/v1/files', {
          method: 'POST',
          headers: {
            Authorization: `Bearer ${openaiApiKey}`
          },
          body: formData
        })
          .then(response => response.json())
          .then(data => {
            console.log(`Uploaded file: ${file}`, data);
          })
          .catch(error => {
            console.error('Error uploading file:', error);
          });
      }
    });
  });
};

uploadFiles();

File: .github/workflows/upload_to_openai.py
import os
import openai

def upload_files():
    openai.api_key = os.environ.get("OPENAI_API_KEY")
    build_dir = "build"

    for file in os.listdir(build_dir):
        file_path = os.path.join(build_dir, file)
        if os.path.isfile(file_path):
            with open(file_path, "rb") as f:
                openai.File.create(
                    file=f,
                    purpose="code_review"
                )
            print(f"Uploaded file: {file}")

if __name__ == "__main__":
    upload_files()